src/main/java/com/google/googleidentity/servlet/RegisterCheckServlet.java

```diff
@@ -47,19 +48,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
     String username = request.getParameter("username");
     String password = request.getParameter("password");
+    String email = request.getParameter("email");
 
     response.setContentType("text/html;charset=utf-8");
 
     if (userDetailsService.getUserByName(username).isPresent()) {
       response.setStatus(HttpStatusCodes.STATUS_CODE_UNAUTHORIZED);
     } else {
-      userDetailsService.addUser(
-          UserDetails.newBuilder().setUsername(username).setPassword(password).build());
+      UserDetails.Builder builder =
```

Comment:
email should be required field, and also need to check if the email has been used in other accounts, otherwise it would be not possible to do link by assertion if there are multiple accounts using the same email.